extensions/amp-story/1.0/amp-story-background-blur.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {setStyles} from '#core/dom/style';
+
+/** @const {number} */
+const CANVAS_SIZE = 3;
+
+/** @const {number} */
+const DURATION_MS = 200;
+
+export class AmpStoryBackgroundBlur {
+  /**
+   * @param {!Window} win
+   * @param {!Element} element
+   */
+  constructor(win, element) {
+    /** @private @const {!Element} */
+    this.element_ = element;
+
+    /** @private @const {!Window} */
+    this.win_ = win;
+
+    /** @private @const {!Element} */
+    this.canvas_ = null;
+  }
+
+  /**
+   * Setup canvas and attach it to the document.
+   */
+  attach() {
+    this.canvas_ = document.createElement('canvas');
+    this.canvas_.width = this.canvas_.height = CANVAS_SIZE;
+    setStyles(this.canvas_, {
+      width: '100%',
+      height: '100%',
+      position: 'absolute',
+      left: 0,
+      top: 0,
+    });
+    this.element_.appendChild(this.canvas_);
+  }
+
+  /**
+   * Update the background to the specified page's background.
+   * @param {!Element} pageElement
+   */
+  update(pageElement) {
+    const fillElement = this.getBackgroundElement_(pageElement);
```

Comment:
Can you add some kind of logging when no image is found? I feel like this is going to be helpful if we have to debug the background image selection